src/test/datascience/raw-kernel/rawKernel.functional.test.ts

```
@@ -0,0 +1,117 @@
+import { IJMPConnection } from '../../../client/datascience/types';
+import { DataScienceIocContainer } from '../dataScienceIocContainer';
+
+// tslint:disable:no-any no-multiline-string max-func-body-length no-console max-classes-per-file trailing-comma
+suite('DataScience raw kernel tests', () => {
+    let ioc: DataScienceIocContainer;
+    let enchannelConnection: IJMPConnection;
+    let connectionFile: string;
+    let kernelProcResult: ObservableExecutionResult<string>;
+    const connectionInfo = {
+        shell_port: 57718,
+        iopub_port: 57719,
+        stdin_port: 57720,
+        control_port: 57721,
+        hb_port: 57722,
+        ip: '127.0.0.1',
+        key: 'c29c2121-d277576c2c035f0aceeb5068',
+        transport: 'tcp',
+        signature_scheme: 'hmac-sha256',
+        kernel_name: 'python3',
+        version: 5.1
+    };
+    setup(async function() {
+        ioc = new DataScienceIocContainer();
+        ioc.registerDataScienceTypes();
+        await ioc.activate();
+        if (ioc.mockJupyter) {
+            // tslint:disable-next-line: no-invalid-this
+            this.skip();
+        } else {
+            enchannelConnection = ioc.get<IJMPConnection>(IJMPConnection);
+
+            // Find our jupyter interpreter
+            const interpreter = await ioc.getJupyterCapableInterpreter();
+            assert.ok(interpreter, 'No jupyter interpreter found');
+            // Start our kernel
+            const execFactory = ioc.get<IPythonExecutionFactory>(IPythonExecutionFactory);
+            const env = await execFactory.createActivatedEnvironment({ interpreter });
+
+            connectionFile = path.join(os.tmpdir(), `tmp_${Date.now()}_k.json`);
+            await fs.writeFile(connectionFile, JSON.stringify(connectionInfo), { encoding: 'utf-8', flag: 'w' });
+
+            // Keep kernel alive while the tests are running.
+            kernelProcResult = env.execObservable(['-m', 'ipykernel_launcher', '-f', connectionFile], {
```

didn't realize there was this ipykernel_launcher. Basically just the same as the  "IPython.kernel" from default kernelspecs?

---

Not sure. This is the command line that the juptyer_client thing you sent me was showing. This allows the test to output the kernel errors though instead of the jupyter_client swallowing them